publish_courses/api.py

```
@@ -0,0 +1,118 @@
+
+logger = logging.getLogger(__name__)
+PC_PERMISSION = settings.PERMISSION_PUBLISH_COURSES
+
+
+class CanvasAPIError(APIException):
+    default_detail = u'Canvas API error'
+
+
+class ProcessSerializer(ModelSerializer):
+    details = JSONField()
+
+    class Meta:
+        model = Process
+        exclude = ('name', 'source')
+
+
+class LTIPermission(BasePermission):
+    def has_permission(self, request, view):
+        return lti_permission_required_check(request, PC_PERMISSION)
+
+
+class SummaryList(ListAPIView):
+    """
+    Return counts of published courses using the GET parameters given
+    :param term_id: The SIS term to count course instances for
+    :param account_id: The SIS school ID to count course instances for
+    :return: JSON response containing the course instance counts
+    """
+    permission_classes = (LTIPermission,)
+
+    def list(self, request, *args, **kwargs):
+        self.term_id = self.request.query_params.get("term_id")
+        self.account_id = self.request.query_params.get("account_id")
+        self.sis_term_id = 'sis_term_id:{}'.format(self.term_id)
+        self.sis_account_id = 'sis_account_id:school:{}'.format(self.account_id)
+        try:
+            total_courses = self._get_courses()
+            state_map = {
+                'published': 'available',
+                'unpublished': 'unpublished',
+                'concluded': 'completed',
+            }
+            summary_counts_by_state = {'total': len(total_courses)}
+            for k, v in state_map.items():
+                summary_counts_by_state[k] = len(
+                    [c for c in total_courses if c['workflow_state'] == v])
+        except Exception as e:
+            logger.exception(
+                "Failed to get published courses summary for term_id={} and "
+                "account_id={}".format(self.term_id, self.account_id))
+            raise CanvasAPIError('There was a problem counting courses. '
```

Agree, we could re-raise the exception with a new message